Add custom emojis, add @holepunchto/emojis dependency

ycharniauski opened the pull request:
Related PR: https://github.com/holepunchto/emojis/pull/6 (should be merged first)

ycharniauski commented on test.js:
```diff
@@ -21,3 +21,25 @@ test('emojis over text', function (t) {
+  t.is(e.toEmoji('holepunch'), '🕳️🥊')
+
+  t.is(e.toShortCode('😂'), 'joy')
+  t.is(e.toShortCode('🥳'), 'keet_party') // should be 'hooray'
```
@mafintosh can you review please?
I added a test and it shows that 🥳 is converting to `keet_party` instead of `hooray`. 
Should it be fixed or should I keep it 'as is'? May be not a big issue. Do we use a backward conversion from icon to shortcode?

If it should be fixed I can suggest extend `toShortCode` and take pointers around 'mid' and see it's icons. If icon is the same as target icon, then I can compare shortcodes and select a shortcode that is not started with 'keet'.

ycharniauski replied:
@mafintosh also not sure about  `t.is(e.toShortCode('bitcoin'), 'keet_bitcoin')`
I added prefix `keet_` to bitcoin because there is no 'bitcoin' emoji in emojis set. But in future it can be added I think. And to avoid conflicts I put a `keet_` prefix.
But may be we will have a weird situation if we convert 'bitcoin price is ...' text to text with emojis. It will bee `keet_bitcoin price is ...` I think. So, there is no 100% backward compatibility. Probably we shouldnt add emoji for `bitcoin` to emojis-index.

mafintosh replied:
i think to fix that you just wanna add them to data before adding the others.

one thing i am realising now though is that this is a public module, so we should prop move the "compat" list to just be hardcoded here or be a .json file here

ycharniauski commented on test.js:
```diff
@@ -21,3 +21,25 @@ test('emojis over text', function (t) {
+  t.is(e.toShortCode('😍'), 'keet_love') // should be 'heart_eyes'
+  t.is(e.toShortCode('😌'), 'relieved')
+  t.is(e.toShortCode('🎶'), 'musical_notes')
+  t.is(e.toShortCode('bitcoin'), 'keet_bitcoin')
```
@mafintosh also not sure about  `t.is(e.toShortCode('bitcoin'), 'keet_bitcoin')`
I added prefix `keet_` to bitcoin because there is no 'bitcoin' emoji in emojis set. But in future it can be added I think. And to avoid conflicts I put a `keet_` prefix.
But may be we will have a weird situation if we convert 'bitcoin price is ...' text to text with emojis. It will bee `keet_bitcoin price is ...` I think. So, there is no 100% backward compatibility. Probably we shouldnt add emoji for `bitcoin` to emojis-index.

mialso replied:
@ycharniauski "slack" emoji sets (aka custom packs) do the same, seems pretty valid since "bitcoin" search will find all prefixes?
![image](https://github.com/holepunchto/emoji-index/assets/8707385/3a7cfbab-500e-4a0b-8e17-f1096e67f0a3)

mialso replied:
I see there is an option to search not only by "short-code": `pray` and `people_hugging` examples found for the `thank` search

mialso replied:
I would expect to have a one2one short-code --- emoji relationship, can't actually get the reason to have a single emoji for multiple shortcodes, otherwise we need to improve the search? to provide "keet_bitcoin" emoji for any "bit" or "coin" string search?

mafintosh replied:
I agree with @mialso here, no need for prefixing

ycharniauski replied:
@mialso I mean that short code should be uniq. It is not about search. For search we are using tags.
```
  {
    label: "folded hands",
    hexcode: "1F64F",
    tags: [ < -- tags, not uniq.
      "ask",
      "hand",
      "high 5",
      "high five",
      "please",
      "pray",
      "thanks" <-- tags included `thanks` word
    ],
    emoji: "🙏",
    text: "",
    type: 1,
    order: 423,
    group: 1,
    subgroup: 19,
    version: 0.6,
    skins: [...],
    shortCodes: [ // <-- there is no `thanks` code here. shortcodes should be uniq
      "folded_hands",
       "pray"
      ] 
  }

```
When you type `thank` slack is searching 'thank' string in tags and also in shortcodes. Even shortcode not includes `thank` string `pray` icon will be shown because `pray` emoji includes`thanks` in tags as I know.

If one day 'bitcoin' emoji will be included to base emojis set then our solution will be broken because we will have 2 emojis with the same ':bitcoin:' shortcode. @holepunchto/emojis will throw exception.
> I would expect to have a one2one short-code --- emoji relationship

exactly, thats why I suggest use name ':keet_bitcoin:'. 
How should code know what icon it needs  render if 2 emojis have the same shortcode 'bitcoin' and text includes `:bitcoin:` ? 
If message includes a emoji as character - then no problems. But If message includes :bitcoin: shortcode then we will have a conflict. I mean, now I can addd just a 'bitcoin' icon because it is a uniq short code.

mialso replied:
I agree, just wanna say the "search" and "tag(key) uniqueness" are separate problems - I mean neither "bitcoin" nor "keet_bitcoin" solves uniqueness problem completely - still need some logic, taking cases if user is allowed to create own tags and upload custom emojis as well

mafintosh commented on package.json:
```diff
@@ -8,6 +8,7 @@
     "raw-index.js"
   ],
   "dependencies": {
+    "@holepunchto/emojis": "^1.0.0",
```
should be a dev dependency

ycharniauski commented:
Please review again, moved @holepunchto/emojis to devDependencies, renamed 'keet_bitcoin' with 'bitcoin', and 'keet_tether' with 'tether'.
Also converted to draft, because @holepunchto/emojis should be merged first

ycharniauski committed:
Add custom emojis

mafintosh committed:
Update raw-emojis-custom.js

mafintosh committed:
Update generate.mjs

ycharniauski committed:
Exclude custom codes from reverse that have conflicts with base emojis codes
